src/mame/capcom/lwings.cpp

```diff
@@ -121,161 +125,23 @@ void lwings_state::avengers_interrupt(int state)
 
 void lwings_state::avengers_protection_w(uint8_t data)
 {
-	int pc = m_maincpu->pc();
+	m_mcu_prot_val = data;
 
-	if (pc == 0x2eeb)
-	{
-		m_param[0] = data;
-	}
-	else if (pc == 0x2f09)
-	{
-		m_param[1] = data;
-	}
-	else if(pc == 0x2f26)
-	{
-		m_param[2] = data;
-	}
-	else if (pc == 0x2f43)
+	if (m_mcu)
 	{
-		m_param[3] = data;
-	}
-	else if (pc == 0x0445)
-	{
-		m_soundstate = 0x80;
-		m_soundlatch->write(data);
+		m_mcu->set_input_line(0, ASSERT_LINE);
+		m_irq_ack_timer->adjust(attotime::from_msec(1000/60));
 	}
```

Comment:
You probably want to either use a latch device, or use a sync barrier so the microcontroller sees the interrupt and updated value at the same time.